pkg/plugin/processor/builtin/js/processor.go

```diff
@@ -0,0 +1,370 @@
+		r := singleRecord{
+			Metadata: make(map[string]string),
+		}
+		// We need to return a pointer to make the returned object mutable.
+		return runtime.ToValue(&r).ToObject(runtime)
+	}
+}
+
+func (p *processor) newFilterRecord(runtime *goja.Runtime) func(goja.ConstructorCall) *goja.Object {
+	return func(call goja.ConstructorCall) *goja.Object {
+		// We need to return a pointer to make the returned object mutable.
+		return runtime.ToValue(&filterRecord{}).ToObject(runtime)
+	}
+}
+
+func (p *processor) newErrorRecord(runtime *goja.Runtime) func(goja.ConstructorCall) *goja.Object {
+	return func(call goja.ConstructorCall) *goja.Object {
+		rec := &errorRecord{}
+		if len(call.Arguments) > 0 {
+			rec.Error = call.Arguments[0].String()
+		}
+		// We need to return a pointer to make the returned object mutable.
+		return runtime.ToValue(rec).ToObject(runtime)
+	}
+}
+
+func (p *processor) jsContentRaw(runtime *goja.Runtime) func(goja.ConstructorCall) *goja.Object {
+	return func(call goja.ConstructorCall) *goja.Object {
+		var r opencdc.RawData
+		if len(call.Arguments) > 0 {
+			r = opencdc.RawData(call.Argument(0).String())
+		}
+		// We need to return a pointer to make the returned object mutable.
+		return runtime.ToValue(&r).ToObject(runtime)
+	}
+}
+
+func (p *processor) jsContentStructured(runtime *goja.Runtime) func(goja.ConstructorCall) *goja.Object {
+	return func(call goja.ConstructorCall) *goja.Object {
+		// TODO accept arguments
+		// We return a map[string]interface{} struct, however because we are
+		// not changing call.This instanceof will not work as expected.
+
+		r := make(map[string]interface{})
+		return runtime.ToValue(r).ToObject(runtime)
+	}
+}
+
+func (p *processor) toJSRecords(runtime *goja.Runtime, recs []opencdc.Record) goja.Value {
+	convertData := func(d opencdc.Data) interface{} {
+		switch v := d.(type) {
+		case opencdc.RawData:
+			return &v
+		case opencdc.StructuredData:
+			return map[string]interface{}(v)
+		}
+		return nil
+	}
+
+	jsRecs := make([]*singleRecord, len(recs))
+	for i, r := range recs {
+		jsRecs[i] = &singleRecord{
+			Position:  r.Position,
+			Operation: r.Operation.String(),
+			Metadata:  r.Metadata,
+			Key:       convertData(r.Key),
+			Payload: struct {
+				Before interface{}
+				After  interface{}
+			}{
+				Before: convertData(r.Payload.Before),
+				After:  convertData(r.Payload.After),
+			},
+		}
+	}
+
+	// we need to send in a pointer to let the user change the value and return it, if they choose to do so
+	return runtime.ToValue(jsRecs)
+}
+
+func (p *processor) toSDKRecords(runtime *goja.Runtime, v goja.Value, recordsCount int) []sdk.ProcessedRecord {
+	var jsRecs []interface{}
+	err := runtime.ExportTo(v, &jsRecs)
+	if err != nil {
+		return p.makeProcessedRecords(
+			sdk.ErrorRecord{Error: cerrors.Errorf("failed exporting JavaScript records to Go values: %w", err)},
+			recordsCount,
+		)
+	}
+
+	out := make([]sdk.ProcessedRecord, len(jsRecs))
```

Comment:
lets make sure this slice is fixed for all the processors, so that the length is not len(records), and that we append to it wach time

Reply:
Done!